EasyVCR/Handlers/VCRHandler.cs

```diff
@@ -67,18 +71,63 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
                     // try to get recorded request
                     var replayInteraction = await FindMatchingInteraction(request);
                     if (replayInteraction == null) throw new VCRException($"No interaction found for request {request.Method} {request.RequestUri}");
+                    if (_validTimeFrame.HasLapsed(replayInteraction.RecordedAt))
+                    {
+                        // matching interaction is expired
+                        switch (_whenExpired)
+                        {
+                            case ExpirationActions.Warn:
+                                // just throw a warning
+                                // will still simulate delay below
+                                Console.WriteLine($"WARNING: Matching interaction is expired.");
```

Comment:
Sure, I can try something like this: https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.warningexception?view=net-6.0

Reply:
The concept at the link above does not work, it is still an exception ultimately, which causes the execution to halt.

I have instead implemented the ability for the end-user to pass in their custom logger to EasyVCR. If a logger is provided, EasyVCR will log the warning to their logger. Otherwise, it will write to console.